Library/Homebrew/download_strategy.rb

```diff
@@ -608,6 +608,59 @@ def fetch_release_metadata
   end
 end
 
+class ScpDownloadStrategy < AbstractFileDownloadStrategy
+  attr_reader :tarball_path, :temporary_path
+
+  def initialize(name, resource)
+    super
+    @tarball_path = HOMEBREW_CACHE/"#{name}-#{version}#{ext}"
+    @temporary_path = Pathname.new("#{cached_location}.incomplete")
+    parse_url_pattern
+  end
+
+  def parse_url_pattern
+    url_pattern = %r{scp://([^@]+@)?([^@:/]+)(:[0-9]+)?/(\S+)}
```

Comment:
This regex is a bit gnarly. Can it e simplified at all? e.g. at least `[0-9]` can be `\d`

Reply:
I'll change `[0-9]` to `\d`.

As far as simplification, I'm not sure how much I can do to change it and still extract the desired parts. Two options:
1. Use whitespace mode and document the regex with comments
2. Put the regex in a constant and test it specifically so people changing it can at least see what behavior they need to maintain (this is already done indirectly via the tests for `fetch` but I'd be happy to move them to make it easier to understand/change)